app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -424,6 +424,9 @@ const Topic = RestModel.extend({
           "details.can_delete": false,
           "details.can_recover": true,
         });
+        if (!User.currentProp("staff")) {
```

Comment:
Can we do this without using `User.currentProp`? We are trying to get away from that. Is `deleted_by` an object with `.staff`? Could we add another argument to pass the current user in if it's not?

Reply:
Excellent suggestion! Yes `deleted_by` is an object with `.staff`. I've made the required changes.